app/src/main/java/com/lagradost/quicknovel/providers/RoyalRoadProvider.kt

```diff
@@ -375,6 +375,18 @@ class RoyalRoadProvider : MainAPI() {
     override suspend fun loadHtml(url: String): String? {
         val response = app.get(url)
         val document = Jsoup.parse(response.text)
-        return document.selectFirst("div.chapter-content")?.html()
+        val styles = document.select("style")
+        val hiddenRegex = Regex("^\\s*(\\..*)\\s*\\{")
```

Comment:
Should be a multiline regex that checks `display: none`, but I couldn't get that to work and cba to play around until kotlin let's me do multiline. (I did try RegexOptions.MULTILINE though, so it either isn't good enough or morely likely the regex I tested in python just doesn't work here fsr)